src/spatial_reaction_systems/spatial_ODE_systems.jl

```diff
@@ -0,0 +1,182 @@
+    num_verts::Int64
+    """The number of species."""
+    num_species::Int64
+    """The values of the parameters which values are tied to vertexes."""
+    vert_ps::Vector{Vector{S}}
+    """Temporary vector. For parameters which values are identical across the lattice, at some point these have to be converted of a length num_verts vector. To avoid re-allocation they are written to this vector."""
+    work_vert_ps::Vector{S}    
+    """For each parameter in vert_ps, its value is a vector with length either num_verts or 1. To know whenever a parameter's value need expanding to the work_vert_ps array, its length needs checking. This check is done once, and the value stored to this array. This field (specifically) is an enumerate over that array."""
+    enum_v_ps_idx_types::Base.Iterators.Enumerate{BitVector}
+    """A vector of pairs, with a value for each species with transportation. The first value is the species index (in the species(::ReactionSystem) vector), and the second is a vector with its transport rate values. If the transport rate is uniform (across all edges), that value is the only value in the vector. Else, there is one value for each edge in the lattice."""
+    transport_rates::Vector{Pair{Int64, Vector{S}}}
+    """A matrix, NxM, where N is the number of species with transportation and M the number of vertexes. Each value is the total rate at which that species leaves that vertex (e.g. for a species with constant diffusion rate D, in a vertex with n neighbours, this value is n*D)."""
+    leaving_rates::Matrix{S}
+    """An (enumerate'ed) iterator over all the edges of the lattice."""
+    enum_edges::T
+    
+    function LatticeDiffusionODEf(ofunc::R, vert_ps::Vector{Vector{S}}, transport_rates::Vector{Pair{Int64, Vector{S}}}, lrs::LatticeReactionSystem) where {R,S}
+        leaving_rates = zeros(length(transport_rates), lrs.num_verts)           # Initialises the leaving rates matrix with zeros.
+        for (s_idx, rates) in enumerate(last.(transport_rates)),
+            (e_idx, e) in enumerate(edges(lrs.lattice))                         # Iterates through all edges, and all transport rates (map from each diffusing species to its rates across edges).
+    
+            leaving_rates[s_idx, e.src] += get_component_value(rates, e_idx)    # Updates the leaving rate for that combination of vertex and species. RHS finds the value of edge "e_idx" in the vector of diffusion rates ("rates").
+        end
+        work_vert_ps = zeros(lrs.num_verts)                                     # Initialises the work vert_ps vector to be empty.
+        enum_v_ps_idx_types = enumerate(length.(vert_ps) .== 1)                 # Creates a Boolean vector whether each vertex parameter need expanding or (and enumerates it, since it always appear in this form).
+        enum_edges = deepcopy(enumerate(edges(lrs.lattice)))                    # Creates an iterator over all the edges. Again, this is always used in the enumerated form. 
```

Comment:
We don't actually got access to `lrs` or `lattice` in the ODE function, so we need to save edges. Since the pure `edges` are never actually used, but always the `enumerate` form, I figured I'd just save the enumerate form

Reply:
changed